content/en/docs/concepts/configuration/secret.md

```diff
@@ -676,6 +676,35 @@ A container using a Secret as a
 Secret updates.
 {{< /note >}}
 
+{{< feature-state for_k8s_version="v1.18" state="alpha" >}}
+
+For clusters that are extensively using secrets (having at least tens of thousands of unique
+Secret to Pod mounts), Kubernetes has an _Immutable Secrets/ConfigMaps_ feature. It allows you
+to mark individual Secrets/ConfigMaps as `Immutable`, which blocks any changes to their data
+and in return:
+- protects you from accidental (or unwanted) updates that could cause applications outages
+- improves performance of your cluster by significantly reducing load on kube-apiserver, by
+closing watches for secrets marked as immutable.
+
+To use it, you need to enable `ImmutableEmphemeralVolumes`
```

Comment:
Possible edit, something like:

To use this feature, enable the [ImmutableEmphemeralVolumes feature gate](...)
and set your Secret or ConfigMap `immutable` field to `true`. For example:

Reply:
done